src/transactions/TransactionFrame.cpp

```diff
@@ -493,8 +683,23 @@ TransactionFrame::processSignatures(ValidationType cv,
 bool
 TransactionFrame::isBadSeq(LedgerTxnHeader const& header, int64_t seqNum) const
 {
-    return seqNum == INT64_MAX || seqNum + 1 != getSeqNum() ||
-           getSeqNum() == getStartingSequenceNumber(header);
+    if (seqNum == INT64_MAX || getSeqNum() == getStartingSequenceNumber(header))
+    {
+        return true;
+    }
+
+    if (mEnvelope.type() == ENVELOPE_TYPE_TX &&
```

Comment:
you don't need to check for `mEnvelope.type() == ENVELOPE_TYPE_TX` here (`getMinSeqNum` takes care of it)